### PR TITLE
envy pr in 2024 (real)

### PR DESCRIPTION
**Pull request by envyaims**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/platinum/joi-18-BitarosBirthday.mdx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+int main() {
+	cin.tie(0)->sync_with_stdio(0);
```
Comment (SansPapyrus683) [inline solutions/platinum/joi-18-BitarosBirthday.mdx:35]: remove if runs w/o

File: solutions/platinum/joi-18-BitarosBirthday.mdx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	int n, m, q;
+	cin >> n >> m >> q;
+	vector<vector<int>> rg(n);
+	for (int i = 0; i < m; i++) {
+		int u, v;
+		cin >> u >> v;
+		rg[--v].push_back(--u);
+	}
+
+	int SQ = 100;  // cutoff for sqrt decomp
```
Comment (SansPapyrus683) [inline solutions/platinum/joi-18-BitarosBirthday.mdx:45]: can you calculate this based on n instead

File: solutions/platinum/joi-18-BitarosBirthday.mdx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	vector<vector<pair<int, int>>> path_sizes(
+	    n);                   // stores the 100 longest paths ending at i
```
Comment (SansPapyrus683) [inline solutions/platinum/joi-18-BitarosBirthday.mdx:47]:
```suggestion
	// stores the 100 longest paths ending at i
	vector<vector<pair<int, int>>> path_sizes(n);

File: solutions/platinum/joi-18-BitarosBirthday.mdx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+		// reset
+		for (int j : from_indicies) { from[j] = -1; }
+	}
+
+	vector<bool> blocked(n);
+	while (q--) {
+		int t, y;
+		cin >> t >> y;
+		--t;
```
Comment (SansPapyrus683) [inline solutions/platinum/joi-18-BitarosBirthday.mdx:80]:
```suggestion
		t--;
```

File: solutions/platinum/joi-18-BitarosBirthday.mdx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		sort(path_sizes[i].rbegin(), path_sizes[i].rend());
+		// pop until sqrt paths
+		while (path_sizes[i].size() > SQ) { path_sizes[i].pop_back(); }
+
+		// reset
+		for (int j : from_indicies) { from[j] = -1; }
+	}
+
+	vector<bool> blocked(n);
+	while (q--) {
```
Comment (SansPapyrus683) [inline solutions/platinum/joi-18-BitarosBirthday.mdx:77]:
```suggestion
	for (int query = 0; query < q; query++) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
love how you submit a pr when progress forms aren't even a thing anymore

**Review by ryanchou-dev — COMMENTED** [review]
¡No puedo creerlo! Estoy muy agradecido 🤩

File: solutions/platinum/joi-18-BitarosBirthday.mdx (hunk 2 above)
Comment (envyaims) [reply solutions/platinum/joi-18-BitarosBirthday.mdx:45]: > can you calculate this based on n instead

did u read the warning

File: solutions/platinum/joi-18-BitarosBirthday.mdx (hunk 1 above)
Comment (envyaims) [reply solutions/platinum/joi-18-BitarosBirthday.mdx:35]: no it doenst

File: solutions/platinum/joi-18-BitarosBirthday.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/platinum/joi-18-BitarosBirthday.mdx]: fail to see how that prevents you from calculating it dynamically

**Review by SansPapyrus683 — APPROVED** [review]
i suppose this is acceptable, show ss of ac

**Comment by envyaims** [comment]
> i suppose this is acceptable, show ss of ac

https://oj.uz/submission/999456

File: solutions/platinum/joi-18-BitarosBirthday.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/platinum/joi-18-BitarosBirthday.mdx]: actually nvm

**Review by ryanchou-dev — APPROVED** [review]
Bueno!

**Pull request merged**